homeassistant/components/zha/repairs.py

```diff
@@ -0,0 +1,67 @@
+"""ZHA repairs for common environmental and device problems."""
+from __future__ import annotations
+
+from universal_silabs_flasher.const import ApplicationType
+from universal_silabs_flasher.flasher import Flasher
+
+from homeassistant.components.homeassistant_yellow import hardware as yellow_hardware
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady, HomeAssistantError
+from homeassistant.helpers import issue_registry as ir
+
+from .core.const import DOMAIN
+
+DISABLE_MULTIPAN_URL_YELLOW = (
+    "https://yellow.home-assistant.io/guides/disable-multiprotocol/"
+)
+DISABLE_MULTIPAN_URL_SKYCONNECT = (
+    "https://skyconnect.home-assistant.io/procedures/disable-multiprotocol/"
+)
+
+
+ISSUE_WRONG_SILABS_FIRMWARE_INSTALLED = "wrong_silabs_firmware_installed"
+
+
+async def probe_silabs_firmware_type(device: str) -> ApplicationType | None:
+    """Probe the running firmware on a Silabs device."""
+    flasher = Flasher(device=device)
```

Comment:
If we already install the flasher, can't we just flash it?

Reply:
I can add a special case for the SkyConnect and the Yellow.

Maybe we can later expand it to a platform where integrations can provide hardware firmware updates? Or have this instead be provided by the SkyConnect/Yellow hardware integrations?

Reply:
Would we download on demand or bundle it with the Core? If we download it, I guess we'd need some type of metadata online/manage releases/versions. Currently the FWs are just as commits in a git repoistory, but that makes it a bit painful to download/figure out what the latest version is.

@puddly already suggested earlier to use GitHub releases. Maybe that would fit the bill for this use case as well :thinking:

Reply:
True, firmwares need to be linked to somehow. Let's keep this as just a repair linking to documentation for now so that it can be merged. We can extend it in the near future to automatically flash if this is feasible.